src/routes/events.js

```diff
@@ -1,12 +1,24 @@
 const { HttpError } = require('@ewarren/serverless-routing');
+const EventS3Model = require('../models/EventS3');
 const EventModel = require('../models/Event');
 const transformEvents = require('../transformers/event');
 
-module.exports = ({ app, s3 }) => {
-  const Event = EventModel(s3);
+module.exports = ({ app, s3, connectToDatabase }) => {
+  app.get('/upcoming', async ({ success, failed, event }) => {
+    const { queryStringParameters } = event;
+    const {
+      v2 = false,
+    } = (queryStringParameters || {});
 
-  app.get('/upcoming', async ({ success, failed }) => {
-    const upcomingEvents = await Event.getUpcomingEvents();
+    var upcomingEvents;
+    if (v2) {
```

Comment:
So this is why I really wanted to avoid trying to make both versions work in the same application, it's adding unnecessary complexity as we have to maintain two API's within the same codebase that rely on different data models. I think in a traditional service where its more devops effort to deploy a brand new service, or there is more legacy code to be concerned with, this would have to be the only way. But Lambda's are basically free and we should treat them as disposable rather than worrying about complex backwards compatibility. 

I'd also caution against using `v2` as a the name of the parameter, and in general using query parameters to version an API, I haven't really seen something like that before and seems like a slippery slope to go down.

I think the easiest way to accomplish a new version is the following,

1/ From the current HEAD on master, checkout a new `v1` branch and push that to remote. This is just in case we ever need to deploy a hotfix to the old version.
2/ Start a new branch off of master for `v2` and change the `serverless.yml` file to have a new base path name `basePath: ${self:custom.stage}-events-v2`. 

Now you can safely make any necessary changes without accidentally breaking the v1 version the website is actively relying on.

Reply:
kk let's go with this plan!